src/python/pants/core_tasks/generate_pants_ini.py

```diff
@@ -0,0 +1,60 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import configparser
+import logging
+import os
+import sys
+
+from pants.base.exceptions import TaskError
+from pants.task.task import Task
+from pants.version import VERSION
+
+
+logger = logging.getLogger(__name__)
+
+
+class GeneratePantsIni(Task):
+  """Generate pants.ini with sensible defaults."""
+
+  PANTS_INI = "pants.ini"
+
+  class CustomConfigParser(configparser.ConfigParser):
+    """We monkey-patch the write() function to allow us to write entries in the style `key: value`, 
+       rather than `key:value` or `key : value`."""
+
+    def write(self, fp):
+      # Refer to https://github.com/jaraco/configparser/blob/294a985b759bddd3505ace54ecf56da39c56a302/src/backports/configparser/__init__.py#L916
+      # for the original source.
+      delimiter = ": "
+      if self._defaults:
+        self._write_section(fp, self.default_section, self._defaults.items(), delimiter)
+      for section in self._sections:
+        self._write_section(fp, section, self._sections[section].items(), delimiter)
+
+  def execute(self):
+    if os.stat(self.PANTS_INI).st_size != 0:
+      raise TaskError("{} is not empty! This goal is only meant for first-time "
+                      "users. Please update config values by directly modifying the file.".format(self.PANTS_INI))
```

Comment:
Perhaps dump the snippet. This would go well with the "Instead of monkeypatching ..." comment above if the full content string were built up once and then either written out when the file is empty or not present or else displayed here for helpfullness.

Reply:
I'm not quite sure what you mean by "dump the snippet"?

Also for the case "when the file is not present", `./pants` does not seem to work when `pants.ini` is missing, so the only case we are trying to cover here is `pants.ini` existing but already being populated.

Reply:
Instead of just saying `Please update config values by directly modifying the file.` and leaving the user hanging, you might show them the update by including the config snippet string you now have down on line 37.